File: Semester 3/Paradygmaty Programowania/paradygmaty/trening/lista1.scala

def poleTrojkata(a:Double, b:Double, c:Double):Double= 
  if(a < 0 || b < 0 || c < 0) throw new Exception("Ujemne parametry")
  else{
    val p = (a+b+c)/2
    if(math.abs(b-c) < a && a < b+c) math.sqrt(p*(p-a)*(p-b)*(p-c))
    else throw new Exception("Bledne parametry")
  }
  
poleTrojkata(-2,-3,4)

File: Semestr 3/Paradygmaty Programowania/paradygmaty/trening/lista1.ml
let poleTrojkata(a, b, c)=
	if(a < 0. || b < 0. || c < 0.) then raise(Failure "Ujemne parametry") 
	else 
		if abs_float(b -. c) < a && a < b +. c  then sqrt(5. *. (5. -. a) *. (5. -. b)*.(5. -. c)) 
		else raise (Failure "Bledne parametry");;

poleTrojkata(2.,3.,4.)